### ISSUE TITLE
Drive letter bug on win32 platforms in TSCache in typescript plugin

### ISSUE THREAD

**Issue by lovettchris** [rollup/plugins#1133]
- Rollup Plugin Name: @rollup/plugin-typescript
- Rollup Plugin Version:  8.3.1
- Rollup Version:  2.70.0
- Operating System (or Browser): Windows 11
- Node Version: v16.14.0
- Link to reproduction _(⚠️ read below)_: 

1. on windows, make a "d:\Temp" folder cd into that folder
2. git clone https://github.com/vtec234/vscode-lean4
3. git checkout infoview-hooks
4. npm install
5. npx lerna bootstrap
6. npx lerna run build

(note the lean4-infoview uses rollup)

### Expected Behavior

Build succeeds on Linux, but fails on Windows.  

### Actual Behavior

Notice the drive letter included in the .rollup.cache path below, which makes for an invalid Windows path.

```
exception has occurred: Error: ENOENT: no such file or directory, mkdir '.rollup.cache\d:\Temp\vtec234\vscode-lean4\lean4-infoview\dist\infoview'
  at Object.mkdirSync (node:fs:1336:3)
    at createFileFolder (D:\Temp\vtec234\vscode-lean4\lean4-infoview\node_modules\@rollup\plugin-typescript\dist\index.js:690:28)
    at TSCache.cacheCode (D:\Temp\vtec234\vscode-lean4\lean4-infoview\node_modules\@rollup\plugin-typescript\dist\index.js:703:9)
    at Object.writeFile (D:\Temp\vtec234\vscode-lean4\lean4-infoview\node_modules\@rollup\plugin-typescript\dist\index.js:750:37)
    at Object.writeFile (D:\Temp\vtec234\vscode-lean4\lean4-infoview\node_modules\typescript\lib\typescript.js:18270:14)
    at printSourceFileOrBundle (D:\Temp\vtec234\vscode-lean4\lean4-infoview\node_modules\typescript\lib\typescript.js:108715:24)
    at emitJsFileOrBundle (D:\Temp\vtec234\vscode-lean4\lean4-infoview\node_modules\typescript\lib\typescript.js:108600:13)
    at emitSourceFileOrBundle (D:\Temp\vtec234\vscode-lean4\lean4-infoview\node_modules\typescript\lib\typescript.js:108523:13)
    at forEachEmittedFile (D:\Temp\vtec234\vscode-lean4\lean4-infoview\node_modules\typescript\lib\typescript.js:108262:34)
    at Object.emitFiles (D:\Temp\vtec234\vscode-lean4\lean4-infoview\node_modules\typescript\lib\typescript.js:108503:9)
Note: Using the "preview" debug extension
```

I have another pull request resolves [one problem](https://github.com/rollup/rollup/pull/4440).

Which unblocks this problem.

**Comment by shellscape** [comment]
Thanks for the issue. Please note the bit in the issue template about acceptable reproductions:

>   🚨 Issues WITHOUT a valid reproduction WILL BE CLOSED!
> 
>   Please provide one by:
>   1. Using the REPL.it plugin reproduction template at https://repl.it/@rollup/rollup-plugin-repro
>   2. Provide a minimal repository link **(Read https://git.io/fNzHA for instructions).**
>      Please use NPM for installing dependencies!
>      These may take more time to triage than the other options.
>   3. Using the Rollup REPL at https://rollupjs.org/repl/

Please give https://git.io/fNzHA a read. The repo you've linked as your reproduction is definitely not minimal. We keep them simple as we're short on help and time is precious.

**Comment by timvahlbrock** [comment]
Can confirm this. The TSCache is just replacing capital drive letters: https://github.com/rollup/plugins/blob/5363f55aa1933b6c650832b08d6a54cb9ea64539/packages/typescript/src/tscache.ts#L19

I will see if I can provide a minimal reproduction repo tomorrow.

**Comment by timvahlbrock** [comment]
Reproduction Repository: https://github.com/timvahlbrock/repro-rollup-ts-lower-case-drive-letters

Created #1141 to fix this.

**Comment by shellscape** [comment]
big thanks to @timvahlbrock and @lovettchris for both submitting PRs to fix this and for helping with this issue
